### PR TITLE
Organizations with balance

### PR DESCRIPTION
**Pull request by coderberry**
Run this after deploy:

```ruby
organization = Organization.new(name: "CodeFund").save(validate: false)

User.with_any_roles("administrator", "advertiser").each do |user|
  next unless user.company_name.present?
  organization = Organization.where(name: user.company_name).first_or_create!
  user.update(organization: organization)
end

Campaign.all.each do |campaign|
  campaign.update(organization_id: campaign.user.organization_id)
end

Creative.all.each do |creative|
  creative.update(organization_id: creative.user.organization_id)
end

User.advertisers.each do |user|
  Impression.where(advertiser_id: user.id).update_all(organization_id: user.organization_id)
end
```

### CODE REVIEW

File: app/controllers/administrator_dashboards_controller.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ class AdministratorDashboardsController < ApplicationController
   before_action :authenticate_user!
 
   def show
-    @active_campaigns = Campaign.active.order(name: :asc)
+    @active_campaigns = Campaign.active.order(name: :asc).limit(9)
```
Comment (hopsoft) [inline app/controllers/administrator_dashboards_controller.rb]: Is this a stub before adding pagination?

File: app/javascript/src/app/controllers/select_date_controller.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import moment from 'moment';
+
+export default class extends Controller {
+  connect() {
+    jQuery(this.element).daterangepicker({
+      singleDatePicker: true,
+      showDropdowns: true,
+    });
+  }
+}
```
Comment (hopsoft) [inline app/javascript/src/app/controllers/select_date_controller.js]: I though this already existed?

File: app/models/creative.rb [hunk 3]
```diff
@@ -99,4 +102,8 @@ def assign_wide_image(blob_id)
   def touch_campaigns
     campaigns.map(&:touch)
   end
+
+  def set_organization
+    self.organization = user.organization
+  end
```
Comment (hopsoft) [inline app/models/creative.rb]: The common behavior for the `has_many` relations of `Organization` should perhaps get rolled into a concern. Some thing like:

```ruby
include Organizable
```

File: app/models/organization.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  # extends ...................................................................
+
+  # includes ..................................................................
+  include Organizations::Developable
+
+  # relationships .............................................................
+  has_many :users
+  has_many :campaigns
+  has_many :creatives
+  has_many :organization_transactions
```
Comment (hopsoft) [inline app/models/organization.rb]: What about `has_many :impressions`?

File: db/migrate/20181220153958_create_organization_transactions.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  def change
+    create_table :organization_transactions do |t|
+      t.bigint :organization_id, null: false
+      t.monetize :amount, null: false, default: 0.0
+      t.string :transaction_type, null: false
+      t.datetime :posted_at, null: false
+      t.text :description
+      t.text :reference
+
+      t.index :organization_id
```
Comment (hopsoft) [inline db/migrate/20181220153958_create_organization_transactions.rb]: I'd also index `transaction_type`

File: app/controllers/administrator_dashboards_controller.rb (hunk 1 above)
Comment (coderberry) [reply app/controllers/administrator_dashboards_controller.rb]: No.. it's just a limit to the admin dashboard. We'll enhance it later

File: app/javascript/src/app/controllers/select_date_controller.js (hunk 2 above)
Comment (coderberry) [reply app/javascript/src/app/controllers/select_date_controller.js]: It did not exist for single dates.

**Pull request merged**